src/Chainweb/Utils.hs

```diff
@@ -1119,3 +1126,22 @@ scurry k a b = k (T2 a b)
 suncurry :: forall a b c. (a -> b -> c) -> T2 a b -> c
 suncurry k (T2 a b) = k a b
 {-# INLINE suncurry #-}
+
+-- -------------------------------------------------------------------------- --
+-- Approximate thread delays
+approximately :: Int -> Prob.GenIO -> IO Int
+approximately k gen = max 0 <$!> sample
+  where
+    sample = (round . (/ 256.0) . head) <$!>
+             Prob.samples 1 (Prob.normal mean sdev) gen
+    mean   = fromIntegral $ k * 256
+    sdev   = mean / 6
+
+
+threadDelayRng :: MVar Prob.GenIO
```

Comment:
In the context of `threadDelay` one wouldn't expect a lot of races due to contention. So maybe an `IORef` with `atomicModifyIORef` would be more efficient here? An `MVar` is certainly the safer option, so it's probably fine.

Reply:
`GenIO` is not safe for concurrent modification, so you have to take the lock. I'm ok with these callers butting heads on this lock, they're going to sleep for a long time soon anyways!